### PR TITLE
Add Dockerfile and improve vscode integration

### PR DESCRIPTION
**Pull request by Veng97**

### CODE REVIEW

File: test/test_vanderpol.cpp [hunk 1]
```diff
@@ -68,7 +68,7 @@ int VanderPol()
 
     for (;;) {
         r = optsolver.step(modelX, r.cmd);
-        auto seq = optsolver.getOptimalSequence();
+        // auto seq = optsolver.getOptimalSequence();
```
Comment (nicolapiccinelli) [inline test/test_vanderpol.cpp:71]: `optsolver.getOptimalSequence` should be kept otherwise we are not really testing its implementation in case of NLMPC

File: test/test_lmpc.cpp [hunk 2]
```diff
@@ -130,7 +130,7 @@ TEST_CASE(
     REQUIRE(optsolver.setExogenuosInputs(mpc::cvec<num_dinputs>::Zero(), {0, pred_hor}));
 
     auto res = optsolver.step(mpc::cvec<num_states>::Zero(), mpc::cvec<num_inputs>::Zero());
-    auto seq = optsolver.getOptimalSequence();
```
Comment (nicolapiccinelli) [inline test/test_lmpc.cpp:133]: `optsolver.getOptimalSequence` should be kept otherwise we are not really testing its implementation in case of LMPC

File: include/mpc/LMPC.hpp [hunk 3]
```diff
@@ -134,18 +134,9 @@ namespace mpc
             const cvec<Tnx> XMax, const cvec<Tnu> UMax, const cvec<Tny> YMax,
             const std::array<int, 2> slice)
         {
-            size_t start = slice[0];
```
Comment (nicolapiccinelli) [inline include/mpc/LMPC.hpp:137]: This was a lazy implementation, could be useful to put these checks in a shared function call

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -4,7 +4,7 @@ if(UNIX)
 endif()
 
 IF("${CMAKE_CXX_COMPILER_ID}" MATCHES "(Apple)?[Cc]lang")
-    set(CMAKE_CXX_FLAGS "-Wunused-command-line-argument -std=c++20 -Wall -Wextra -Wfatal-errors -Werror=return-type -g -O0 -fno-pie")
+    set(CMAKE_CXX_FLAGS "-Wunused-command-line-argument -std=c++20 -Wall -Wextra -Wfatal-errors -Werror=return-type -g -O0 -fpie")
```
Comment (nicolapiccinelli) [inline test/CMakeLists.txt:7]: Why this change?

Comment (Veng97) [reply test/CMakeLists.txt:7]: Inside the Dockerfile i uploaded there is an option to enable compilation with Clang, which in some cases gives detailed compile diagnostics. I also installed clang-tidy within the image to help with formatting and code style later on.

To enable Clang compilation, you can unsilence the environment variables in the bottom of the Dockerfile:
...
# # (Optional) Set Clang as the default compiler
# ENV CC=/usr/bin/clang
# ENV CXX=/usr/bin/clang++
...

and make a clean rebuild.

When building with Clang this error occurs:

`Linking CXX executable ../../bin/test_cases_dynamic
/usr/bin/ld: CMakeFiles/test_cases_dynamic.dir/test_discrete_lti_siso.cpp.o: relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIE
/usr/bin/ld: failed to set dynamic section sizes: bad value
clang: fatal error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [test/CMakeFiles/test_cases_dynamic.dir/build.make:148: ../bin/test_cases_dynamic] Error 1
make[1]: *** [CMakeFiles/Makefile2:912: test/CMakeFiles/test_cases_dynamic.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....`

Comment (nicolapiccinelli) [reply test/CMakeLists.txt]: Gotcha, that's fine!

File: include/mpc/LMPC.hpp (hunk 3 above)
Comment (nicolapiccinelli) [reply include/mpc/LMPC.hpp:137]: @Veng97 if we solve this then we can merge
